model/span.go

```diff
@@ -49,19 +49,19 @@ func (s *Span) Hash(w io.Writer) (err error) {
 }
 
 // HasSpanKind returns true if the span has a `span.kind` tag set to `kind`.
-func (s *Span) HasSpanKind(kind ext.SpanKindEnum) bool {
-	if tag, ok := KeyValues(s.Tags).FindByKey(string(ext.SpanKind)); ok {
-		return tag.AsString() == string(kind)
+func (s *Span) HasSpanKind(kind trace.SpanKind) bool {
+	if tag, ok := KeyValues(s.Tags).FindByKey("span.kind"); ok {
```

Comment:
> "span.kind"

instead of using a string, it's better to define a (private) constant, e.g. `const keySpanKind = "span.kind"`